nodebuilder/module.go

```diff
@@ -19,6 +19,18 @@ import (
 )
 
 func ConstructModule(tp node.Type, network p2p.Network, cfg *Config, store Store) fx.Option {
+	// NOTE: we construct keyring signer before constructing node for easier UX
+	// as having keyring-backend set to `file` prompts user for password.
+	log.Infow("Accessing keyring...")
+	ks, err := store.Keystore()
+	if err != nil {
+		fx.Error(err)
+	}
+	signer, err := state.KeyringSigner(cfg.State, ks, network)
```

Comment:
Why not to provide this?

Reply:
```
// NOTE: we construct keyring signer before constructing node for easier UX
	// as having keyring-backend set to `file` prompts user for password.
```

Reply:
Ok, pls move this Note to the KeyringSigner constructor. Otherwise, the connection between the noe and constructor are not obvious

Reply:
ok pls check now